src/psyclone/transformations.py

```diff
@@ -2262,3 +2262,82 @@ def apply(self, sched):
         schedule.addchild(data_dir, index=0)
 
         return schedule, keep
+
+
+class ACCRoutineTrans(Transformation):
+    '''
+    Transform a kernel subroutine through the addition of a "!$acc routine"
+    directive (causing it to be compiled for the OpenACC accelerator device).
+    For example:
+
+    >>> from psyclone.parse import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api = "gocean1.0"
+    >>> filename = "nemolite2d_alg.f90"
+    >>> ast, invokeInfo = parse(filename, api=api, invoke_name="invoke")
```

Comment:
Can we drop the invoke_name argument here as it defaults to "invoke"?

Reply:
Oh yes; done.